src/providers/settings.ts

```diff
@@ -30,73 +28,39 @@ export class Settings {
     this._defaults = defaults;
   }
 
-  _mergeDefaults(defaults: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-      for (let k in defaults) {
-        if (!(k in this.settings)) {
-          this.settings[k] = defaults[k];
-        }
+  _mergeDefaults(defaults: any): void {
+    for (let k in defaults) {
+      if (!(k in this.settings)) {
+        this.settings[k] = defaults[k];
       }
-      this.setAll(this.settings).then(result => {
-        resolve(result);
-      }).catch(error => {
-        reject(error);
-      });
-    });
+    }
+    this.setAll(this.settings);
   }
 
-  merge(settings: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-      for (let k in settings) {
-        this.settings[k] = settings[k];
-      }
-      this.save().then(result => {
-        resolve(result);
-      }).catch(error => {
-        reject(error);
-      });
-    })
+  merge(settings: any): void {
+    for (let k in settings) {
+      this.settings[k] = settings[k];
+    }
+    this.save();
   }
 
-  setValue(key: string, value: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-      this.settings[key] = value;
-      this.storage.set(Constants.STORAGE_KEY_SETTINGS, this.settings).then(result => {
-        resolve(result);
-      }).catch(error => {
-        reject(error);
-      });
-    })
+  setValue(key: string, value: any): void {
+    this.settings[key] = value;
+    this.storage.set(Constants.STORAGE_KEY_SETTINGS, this.settings);
   }
 
-  setAll(value: any): Promise<any> {
-    return new Promise((resolve, reject) => {
-      this.storage.set(Constants.STORAGE_KEY_SETTINGS, value).then(result => {
-        resolve(result);
-      }).catch(error => {
-        reject(error);
-      });
-    });
+  setAll(value: any): void {
+    this.storage.set(Constants.STORAGE_KEY_SETTINGS, value);
   }
 
-  getValue(key: string): Promise<any> {
-    return new Promise((resolve, reject) => {
-      this.storage.get(Constants.STORAGE_KEY_SETTINGS).then(result => {
-        resolve(result);
-      }).catch(error => {
-        reject(error);
-      });
-    });
+  getValue(key: string): any {
+    this.storage.get(Constants.STORAGE_KEY_SETTINGS).then(result => {
```

Comment:
You should return this.storage.get and then remove the `then`. Then you are returning the promise, so there's no need to have a then here. Same for the rest of these. This is what it should look like:

```javascript
return this.storage.get(Constants.STORAGE_KEY_SETTINGS)
```